charts/vela-core/templates/defwithtemplate/config-image-registry.yaml

```diff
@@ -0,0 +1,75 @@
+# Code generated by KubeVela templates. DO NOT EDIT. Please edit the original cue file.
+# Definition source cue file: vela-templates/definitions/internal/config-image-registry.cue
+apiVersion: core.oam.dev/v1beta1
+kind: ComponentDefinition
+metadata:
+  annotations:
+    custom.definition.oam.dev/name.config.oam.dev: Image Registry
+    definition.oam.dev/description: Config information to authenticate image registry
+  labels:
+    custom.definition.oam.dev/catalog.config.oam.dev: velacore-config
+    custom.definition.oam.dev/multi-cluster.config.oam.dev: "true"
+    custom.definition.oam.dev/type.config.oam.dev: image-registry
+  name: config-image-registry
+  namespace: {{ include "systemDefinitionNamespace" . }}
+spec:
+  schematic:
+    cue:
+      template: |
+        import (
+        	"encoding/base64"
+        	"encoding/json"
+        )
+
+        output: {
+        	apiVersion: "v1"
+        	kind:       "Secret"
+        	metadata: {
+        		name:      parameter.name
+        		namespace: "vela-system"
+        		labels: {
+        			"config.oam.dev/catalog":       "velacore-config"
+        			"config.oam.dev/type":          "image-registry"
+        			"config.oam.dev/multi-cluster": "true"
+        			if parameter.accountAuth != _|_ {
+        				"config.oam.dev/identifier": parameter.accountAuth.registry
+        			}
+        			if parameter.noAuth != _|_ {
+        				"config.oam.dev/identifier": parameter.noAuth.registry
+        			}
+        		}
+        	}
+        	type: "kubernetes.io/dockerconfigjson"
+        	stringData: {
+        		if parameter.accountAuth != _|_ {
+        			".dockerconfigjson": json.Marshal({
+        				auths: "\(parameter.accountAuth.registry)": {
+        					username: parameter.accountAuth.username
+        					password: parameter.accountAuth.password
+        					email:    parameter.accountAuth.email
+        					auth:     base64.Encode(null, (parameter.accountAuth.username + ":" + parameter.accountAuth.password))
+        				}
+        			})
+        		}
+        		if parameter.noAuth != _|_ {
+        			registry: parameter.noAuth.registry
+        		}
+        	}
+        }
+        parameter: {
```

Comment:
```
registry: string
accountAuth?: {
        // +usage=Image registry username.
        username: string
        // +usage=Image registry password.
        password: string
 }
```

Reply:
How did you know the type for an image registry? With this style, it's not friendly to display different types for the image registry in VelaUX.